tests/integration/index/query_performance_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package index
+
+import (
+	"fmt"
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func generateDocs(count int) []any {
+	result := make([]any, 0, count)
+	for i := 0; i < count; i++ {
+		result = append(result, testUtils.CreateDoc{
+			CollectionID: 0,
+			Doc: fmt.Sprintf(`{
+				"name": "name-%d",
+				"age":  %d,
+				"email":  "email%d@gmail.com"
+			}`, i, i%100, i),
+		})
+	}
+	return result
+}
+
+func TestQueryPerformance_Simple(t *testing.T) {
+	const benchReps = 3
+	const numDocs = 500
+
+	docs := generateDocs(numDocs)
+
+	const req = `query {
+		User(filter: {age: {_eq: 33}}) {
+			name
+			age
+			email
+			verify
+		}
+	}`
+
+	var benchResRegular testUtils.BenchmarkResult
+	test1 := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{Schema: `
+				type User {
+					name: String
+					age: Int
+					email: String
+					verify: Boolean
+				}
+			`},
+			docs,
+			testUtils.Benchmark{
+				Reps:         benchReps,
+				Action:       testUtils.Request{Request: req},
+				FocusClients: []testUtils.ClientType{testUtils.GoClientType},
+				Result:       &benchResRegular,
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test1)
+
+	var benchResIndexed testUtils.BenchmarkResult
+	test2 := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{Schema: `
+				type User {
+					name: String 
+					age: Int @index
+					email: String
+					verify: Boolean
+				} 
+			`},
+			docs,
+			testUtils.Benchmark{
+				Reps:   benchReps,
+				Action: testUtils.Request{Request: req},
+				Result: &benchResIndexed,
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test2)
+
+	for dbt, regularVal := range benchResRegular.ElapsedTime {
```

Comment:
todo: This should be done within the `Benchmark` action, doing it here escapes the framework and will increase maintenance costs in the long run if we keep doing this.

I think you have also missused the framework somewhat by including two `testUtils.TestCase`s in the same test.  You should be able to remove the need for this by reworking the `Benchmark` action, perhaps so that the test case looks something like:
```
func TestQueryPerformance_Simple(t *testing.T) {
    test := testUtils.TestCase{
	    Actions: []any{
		    testUtils.SchemaUpdate{
			    Schema: `
				    type Users {
					    name: String
					    age: Int
					    email: String
					    verify: Boolean
				    }
			    `,
		    },
		    testUtils.SchemaUpdate{
			    Schema: `
				    type IndexedUsers {
					    name: String
					    age: Int @index
					    email: String
					    verify: Boolean
				    }
			    `,
		    },
		    testUtils.GenerateDocs{
                Count: 500,
            },
		    testUtils.Benchmark{
			    Reps:   benchReps,
			    Base: testUtils.Request{Request: "..."},
			    Comparee: testUtils.Request{Request: "..."}, // Comparee is not a word, try and find a better one :D
			    Factor: 5,
		    },
	    },
    }

    testUtils.ExecuteTestCase(t, test)
}
```

Reply:
Once we are happy with the test action, I would want to know how expensive it is, and I would want more of them.  At the moment we only really have a single test that covers the behaviour that the users care most about.